custom_components/healthchecksio/__init__.py

```diff
@@ -69,27 +82,36 @@ async def async_setup_entry(hass, config_entry):
 class HealthchecksioData:
     """This class handle communication and stores the data."""
 
-    def __init__(self, hass, api_key, check):
+    def __init__(self, hass, api_key, check, self_hosted, site_root, ping_endpoint):
         """Initialize the class."""
         self.hass = hass
         self.api_key = api_key
         self.check = check
+        self.self_hosted = self_hosted
+        self.site_root = site_root if self_hosted else OFFICIAL_SITE_ROOT
+        self.ping_endpoint = ping_endpoint
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     async def update_data(self):
         """Update data."""
         Logger("custom_components.healthchecksio").debug("Running update")
         # This is where the main logic to update platform data goes.
         try:
-            session = async_get_clientsession(self.hass)
+            verify_ssl = not self.self_hosted or self.site_root.startswith("https")
+            session = async_get_clientsession(self.hass, verify_ssl)
             headers = {"X-Api-Key": self.api_key}
             async with async_timeout.timeout(10, loop=asyncio.get_event_loop()):
                 data = await session.get(
-                    "https://healthchecks.io/api/v1/checks/", headers=headers
+                    f"{self.site_root}/api/v1/checks/", headers=headers
                 )
                 self.hass.data[DOMAIN_DATA]["data"] = await data.json()
 
-                await session.get(f"https://hc-ping.com/{self.check}")
+                if self.self_hosted:
+                    check_url = f"{self.site_root}/{self.ping_endpoint}/{self.check}"
+                else:
+                    check_url = f"https://hc-ping.com/{self.check}"
+                await asyncio.sleep(0.00001)
```

Comment:
I assume you put that in there so the server gets time between the ping and the check, but is this needed and is `0.00001` enough if it's needed?

Reply:
Well, I tried to elaborate on this in the wall of text above (comment). When using the local instance I get the `[Errno 54] Connection reset by peer` mentioned in #4, when adding these sleeps they do not occur. When using such a low value it does work 90% of the time, using a longer sleep duration would probably increase this even more, but it was good enough for me. It may be a good Idea to add a comment to those lines explaining the reason they exist though.

Reply:
You certainly did :see_no_evil: 
Let's set this to `1`, it does not block anything so it does not matter.
Just add an inline comment like "# needed for self-hosted instances"

Reply:
Done